plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -0,0 +1,205 @@
+                                    snapshotSequenceNumber,
+                                    ByteBuffer.wrap(sketch.toByteArray()),
+                                    ZSTD,
+                                    ImmutableMap.of(APACHE_DATASKETCHES_THETA_V1_NDV_PROPERTY, Long.toString(ndvEstimate))));
+                        });
+
+                writer.finish();
+                return new GenericStatisticsFile(
+                        snapshotId,
+                        path,
+                        writer.fileSize(),
+                        writer.footerSize(),
+                        writer.writtenBlobsMetadata().stream()
+                                .map(GenericBlobMetadata::from)
+                                .collect(toImmutableList()));
+            }
+            catch (IOException exception) {
+                throw new UncheckedIOException(exception);
+            }
+        }
+        catch (Throwable throwable) {
+            closeAllSuppress(throwable, () -> fileIO.deleteFile(path));
+            throw throwable;
+        }
+    }
+
+    private void copyRetainedStatistics(
+            FileIO fileIO,
+            StatisticsFile previousStatisticsFile,
+            Set<Integer> validFieldIds,
+            Set<Integer> columnsWithNewNdvSketches,
+            PuffinWriter writer)
+    {
+        boolean anythingRetained = previousStatisticsFile.blobMetadata().stream()
+                .anyMatch(blobMetadata -> isBlobRetained(blobMetadata.type(), blobMetadata.fields(), validFieldIds, columnsWithNewNdvSketches));
+
+        if (anythingRetained) {
+            try (PuffinReader reader = Puffin.read(fileIO.newInputFile(previousStatisticsFile.path()))
+                    .withFileSize(previousStatisticsFile.fileSizeInBytes())
+                    .withFooterSize(previousStatisticsFile.fileFooterSizeInBytes())
+                    .build()) {
+                List<BlobMetadata> retained = reader.fileMetadata().blobs().stream()
+                        .filter(blobMetadata -> isBlobRetained(blobMetadata.type(), blobMetadata.inputFields(), validFieldIds, columnsWithNewNdvSketches))
+                        .collect(toImmutableList());
+                for (Pair<BlobMetadata, ByteBuffer> read : reader.readAll(retained)) {
+                    String compressionCodec = read.first().compressionCodec();
+                    writer.add(new Blob(
+                            read.first().type(),
+                            read.first().inputFields(),
+                            read.first().snapshotId(),
+                            read.first().sequenceNumber(),
+                            read.second(),
+                            // TODO Allow PuffinReader to read without decompression
+                            compressionCodec == null
+                                    ? null
+                                    : Enums.getIfPresent(PuffinCompressionCodec.class, compressionCodec).or(ZSTD),
```

Comment:
How about using `PuffinCompressionCodec.forName()`? The `compressionCodec` seems lowercase as far as I read `PuffinWriter` and tested locally.